File: skill_sdk/tests/fixtures/js-project/src/core/Handler.js
import { Helper } from './Helper.js';

/**
 * Processes incoming API requests. Mirrors Handler in the Java fixture.
 */
export class Handler {
  /**
   * @param {import('./DataProcessor.js').DataProcessor} processor
   */
  constructor(processor) {
    /** @type {import('./DataProcessor.js').DataProcessor} */
    this._processor = processor;
  }

  /**
   * @param {string} payload
   * @returns {string}
   */
  processRequest(payload) {
    console.log(`Processing request: ${payload}`);

    const helper = new Helper(this._processor);
    return helper.handleRequest(payload);
  }

  /**
   * @returns {boolean}
   */
  healthCheck() {
    return this._processor !== null;
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/core/Handler.java
package com.fixture.core;

/**
 * Processes incoming API requests. Mirrors Handler in handler.go.
 */
public class Handler {

    private final DataProcessor processor;

    public Handler(DataProcessor processor) {
        this.processor = processor;
    }

    public String processRequest(String payload) {
        System.out.println("Processing request: " + payload);

        Helper helper = new Helper(processor);
        return helper.handleRequest(payload);
    }

    public boolean healthCheck() {
        return processor != null;
    }
}
